src/gui/qgscolorwidgets.cpp

```diff
@@ -868,17 +860,20 @@ void QgsColorBox::paintEvent( QPaintEvent *event )
   painter.drawImage( QPoint( mMargin, mMargin ), *mBoxImage );
 
   //draw cross lines
-  const double xPos = mMargin + ( width() - 2 * mMargin - 1 ) * static_cast<double>( xComponentValue() ) / static_cast<double>( valueRangeX() );
-  const double yPos = mMargin + ( height() - 2 * mMargin - 1 ) - ( height() - 2 * mMargin - 1 ) * static_cast<double>( yComponentValue() ) / static_cast<double>( valueRangeY() );
+  const double h = height();
+  const double w = width();
+  const int margin = static_cast<int>( mMargin );
+  const int xPos = static_cast<int>( mMargin + ( w - 2 * mMargin - 1 ) * static_cast<double>( xComponentValue() ) );
+  const int yPos = static_cast<int>( mMargin + ( h - 2 * mMargin - 1 ) - ( h - 2 * mMargin - 1 ) * static_cast<double>( yComponentValue() ) );
 
   painter.setBrush( Qt::white );
   painter.setPen( Qt::NoPen );
 
-  painter.drawRect( xPos - 1, mMargin, 3, height() - 2 * mMargin - 1 );
-  painter.drawRect( mMargin, yPos - 1, width() - 2 * mMargin - 1, 3 );
+  painter.drawRect( xPos - 1, mMargin, 3, height() - 2 * margin - 1 );
```

Comment:
instead of truncating to int here, we should probably keep the double calculations and then use the drawRect( QRectF( ... ) ) method.  That should give us more hi-dpi accuracy.

Reply:
Done 99ab223c518